rpc/jsonrpc_state_server.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpc
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ava-labs/avalanchego/trace"
+)
+
+type stateReader interface {
+	Tracer() trace.Tracer
+	ReadState(ctx context.Context, keys [][]byte) ([][]byte, []error)
+}
+
+type StateRequest struct {
```

Comment:
Make this request/response type consistent with the name of the api (so if we go with `GetStateKey`, this should be a `GetStateKeyRequest`).